lib/spack/spack/test/conftest.py

```diff
@@ -145,70 +145,84 @@ def commit(message):
         )
         commit_counter += 1
 
-    with working_dir(repo_path):
-        git("init")
+    def _make(branch_name=None):
+        branch_name = branch_name or "1.x"
+        with working_dir(repo_path):
+            git("init")
 
-        git("config", "user.name", "Spack")
-        git("config", "user.email", "spack@spack.io")
+            git("config", "user.name", "Spack")
+            git("config", "user.email", "spack@spack.io")
```

Comment:
I wouldn't change this to make a mock git repo with a feature branch. This fixture is brittle, and requires external tools like `git` so I'd avoid it if possible in unit tests. What I'd do instead is to add further tests in `test/spec_syntax.py` and check that a given string literal is tokenized as we expect. 

Some examples of tests like that are:
https://github.com/spack/spack/blob/3a565c66e9fc0bd5050faf7d07834e20a23b458f/lib/spack/spack/test/spec_syntax.py#L409-L419

For instance, I see that on `develop` the literal `zlib@git.foo/bar` is incorrectly parsed as:
```python
>>> for token in spack.parser.tokenize("zlib@git.foo/bar"):
...     print(token)
... 
(UNQUALIFIED_PACKAGE_NAME, zlib)
(VERSION, @git.foo)
(DAG_HASH, /bar)
```
while in this branch is correctly:
```python
>>> for token in spack.parser.tokenize("zlib@git.foo/bar"):
...     print(token)
... 
(UNQUALIFIED_PACKAGE_NAME, zlib)
(GIT_VERSION, @git.foo/bar)
```

Reply:
I agree that this is a good targeted test.

I felt that other behaviors like lookup etc. should also be tested, and this covered that. I scaled it back to a smaller test: `test_git_branch_with_slash`. Let me know if that seems sufficient and focused.

Reply:
Thanks @scheibelp !